test/e2e/garbage_collector.go

```diff
@@ -719,4 +755,52 @@ var _ = framework.KubeDescribe("Garbage collector", func() {
 			framework.Failf("failed to wait for all pods to be deleted: %v", err)
 		}
 	})
+
+	It("should delete jobs and pods created by cronjob", func() {
+		framework.SkipIfMissingResource(f.ClientPool, CronJobGroupVersionResource, f.Namespace.Name)
+
+		By("Create the cronjob")
+		cronJob := newCronJob("simple", "*/1 * * * ?")
+		cronJob, err := f.ClientSet.BatchV2alpha1().CronJobs(f.Namespace.Name).Create(cronJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Wait for the CronJob to create new Job")
+		err = wait.PollImmediate(500*time.Millisecond, 2*time.Minute, func() (bool, error) {
+			jobs, err := f.ClientSet.Batch().Jobs(f.Namespace.Name).List(metav1.ListOptions{})
+			if err != nil {
+				return false, fmt.Errorf("Failed to list jobs: %v", err)
+			}
+			return len(jobs.Items) > 0, nil
+		})
+		if err != nil {
+			framework.Failf("Failed to wait for the CronJob to create some Jobs: %v", err)
+		}
+
+		By("Delete the cronjob")
+		if err := f.ClientSet.BatchV2alpha1().CronJobs(f.Namespace.Name).Delete(cronJob.Name, getNonOrphanOptions()); err != nil {
```

Comment:
This could be fixed in another PR. OrphanDependents field is deprecated, set propagationPolicy to [Background](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L364) instead.